extensions/amp-story/1.0/amp-story-render-service.js

```diff
@@ -46,5 +62,15 @@ export class AmpStoryRenderService {
         return storyEl.signals().whenSignal(CommonSignals.LOAD_END);
       });
     });
+
+    return this.timer_
+      .timeoutPromise(
+        LOAD_TIMEOUT,
+        whenReadyPromise,
+        `Render timeout waiting for service AmpStoryRenderService to be ready.`
+      )
+      .catch((reason) => {
```

Comment:
Nit: Thinking about this, this way will catch any rejection error that happen in the `whenReadyPromise`, too. It might be better to do a race. Up to you.

```js
return Promise.race([
  whenReadyPromise,
  this.timer_.delay(LOAD_TIMEOUT)
])
```

Reply:
Sorry, just making sure I'm getting it right. Did you mean:
```javascript
return Promise.race([
  whenReadyPromise,
  this.timer_.promise(LOAD_TIMEOUT)
])
```
?

With `timer_.delay(callback, timeout)`, it runs a callback after timeout and returns a number or string. with `timer.promise(timeout)` it returns a promise that resolves after the timeout. I think we want the later if we want to follow the new approach.

Also, even in this approach, wouldn't it still catch any rejection error that happens in `whenReadyPromise`? At least if it rejects before `timer.promise(timeout)`